### PR TITLE
chore(boostrapping): bootstrapping tests, samples, and docs

### PR DESCRIPTION
**Pull request by bcoe**
This stubs out tests for the library, so that we have a bit more of a starting point.

### CODE REVIEW

File: samples/quickstart.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+  // Create the reCAPTCHA client library.
+  const {
+    RecaptchaEnterpriseServiceV1Beta1Client,
+  } = require('@google-cloud/recaptcha-enterprise');
+  const client = new RecaptchaEnterpriseServiceV1Beta1Client();
+
+  // format the path to the project, projectNumber should be
+  // provided.
+  const formattedParent = client.projectPath(projectNumber);
```
Comment (bcoe) [inline samples/quickstart.js]: I ultimately couldn't get a simple integration test running, for a couple of reason's documented here:

https://github.com/googleapis/nodejs-recaptcha-enterprise/issues/12

I'd advocate landing this as a starting point, and we can come back and add more samples now that we have a foundation in place.

File: README.md [hunk 2]
```diff
@@ -52,6 +52,47 @@ npm install @google-cloud/recaptcha-enterprise
...
 
 
+### Using the client library
+
+```javascript
+/**
+ * Instantiate the reCAPTCHA Enterprise client and make request.
+ * @param {string} projectNumber The project number to use when making requests.
+ */
+async function quickstart(projectNumber) {
```
Comment (JustinBeckwith) [inline README.md]: So `projectNumber` and `projectId` are very different things.  I am mostly sure you want `projectId` here.

Comment (JustinBeckwith) [inline README.md]: You want an outer function called `main`, and an inner function called `quickstart`:
https://github.com/googleapis/nodejs-asset/blob/master/samples/quickstart.js

File: README.md [hunk 3]
```diff
@@ -52,6 +52,47 @@ npm install @google-cloud/recaptcha-enterprise
...
+
+  // Create the reCAPTCHA client library.
+  const {
+    RecaptchaEnterpriseServiceV1Beta1Client,
+  } = require('@google-cloud/recaptcha-enterprise');
+  const client = new RecaptchaEnterpriseServiceV1Beta1Client();
+
+  // format the path to the project, projectNumber should be
+  // provided.
+  const formattedParent = client.projectPath(projectNumber);
```
Comment (JustinBeckwith) [inline README.md]: Just get the `projectId` using `const projectId = await client.getProjectId();` and lose the parameter

File: package-lock.json [hunk 4]
```diff
@@ -13,6 +13,48 @@
         "@babel/highlight": "^7.0.0"
```
Comment (JustinBeckwith) [inline package-lock.json]: Please `.gitignore` the lock file :)

File: package.json [hunk 5]
```diff
@@ -29,24 +29,29 @@
     "lodash.merge": "^4.6.0"
   },
   "devDependencies": {
-    "mocha": "^5.2.0",
-    "through2": "^2.0.3",
+    "chai": "^4.2.0",
```
Comment (JustinBeckwith) [inline package.json]: We generally add this to `samples/package.json`

File: samples/.eslintrc [hunk 6]
```diff
@@ -0,0 +1,5 @@
+---
+rules:
+  no-console: off
+  no-process-exit: off
```
Comment (JustinBeckwith) [inline samples/.eslintrc]: Do we need `no-process-exit`?

File: samples/quickstart.js [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+  const request = {
+    parent: formattedParent,
+    assessment: assessment,
+  };
+
+  await client.createAssessment(request);
+}
+// [END recaptcha_enterprise_quickstart]
+
+const args = process.argv.slice(2);
```
Comment (JustinBeckwith) [inline samples/quickstart.js]: So we are inconsistent here across the bard.  I have just been doing `quickstart(...process.argv.slice(2))`.  

DO we need the catch and the explicit exit?

File: samples/system-test/.eslintrc [hunk 8]
```diff
@@ -0,0 +1,5 @@
+---
+env:
+  mocha: true
+rules:
+  node/no-extraneous-require: off
```
Comment (JustinBeckwith) [inline samples/system-test/.eslintrc]: Can we just use a `test/` directory instead of `system-test/`?

File: samples/system-test/samples.test.js [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+ */
+
+'use strict';
+
+const {assert} = require('chai');
+
+const path = require('path');
+const cp = require('child_process');
+const execSync = cmd => cp.execSync(cmd, {encoding: 'utf-8'});
+const cwd = path.join(__dirname, '..');
```
Comment (JustinBeckwith) [inline samples/system-test/samples.test.js]: Technically I don't think the cwd here needs to be explicit.

File: samples/system-test/samples.test.js [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+
+describe('Quickstart', () => {
+  it('should run quickstart', async () => {
+    // TODO: work on setting up a proper integration test.
+    try {
```
Comment (JustinBeckwith) [inline samples/system-test/samples.test.js]: Can we use `assert.throws`?  If this doesn't throw in some weird case, wouldn't we want the test to fail?

File: samples/quickstart.js (hunk 7 above)
Comment (bcoe) [reply samples/quickstart.js]: without the catch, the exception happens as an unhandledRejection, and the program actually ends up exiting with the code `0`; this was causing issues sprucing up old unit tests.

File: samples/system-test/samples.test.js (hunk 9 above)
Comment (bcoe) [reply samples/system-test/samples.test.js]: just cargo culting from our other tests, will remove.

**Pull request merged**